Tools/clinic/clinic.py

```diff
@@ -1959,7 +1993,10 @@ def render_function(
             converter.set_template_dict(template_dict)
 
         f.return_converter.render(f, data)
-        template_dict['impl_return_type'] = f.return_converter.type
+        if f.kind is SETTER:
+            template_dict['impl_return_type'] = 'int'
```

Comment:
https://github.com/python/cpython/pull/112922/files#r1421696103
ditto

Reply:
I think this is fine; we do similar stuff for `__init__` functions. Perhaps a comment could be helpful, though.

```suggestion
            # All setters return an int.
            template_dict['impl_return_type'] = 'int'
```